src/ReTestItems.jl

```diff
@@ -561,9 +561,19 @@ function include_testfiles!(project_name, projectfile, paths, shouldrun, report:
     subproject_root = nothing  # don't recurse into directories with their own Project.toml.
     root_node = DirNode(project_name; report, verbose=true)
     dir_nodes = Dict{String, DirNode}()
-    # setups is populated in store_test_item_setup when we expand a @testsetup
+    # setup_channel is populated in store_test_setup when we expand a @testsetup
     # we set it below in tls as __RE_TEST_SETUPS__ for each included file
-    setups = Dict{Symbol, TestSetup}()
+    setup_channel = Channel{Pair{Symbol, TestSetup}}(Inf)
+    setup_task = @spawn begin
```

Comment:
So it seems like we could possibly use the Channel-bound-to-a-task functionality that I've never actually used here? i.e. keep `setups` being created outside, then do a Channel-task to iterate until empty and then we don't need the `fetch` down below?

This might be a very naive comment and even if the channel-task thing can work, I really don't mind keeping things as-is. It's just I've never actually used the Channel-task-bind thing and it feels like this is an opporunity.

Reply:
hmm, i'm not sure i quite follow

we could define the setups outside, like
```diff
+    setups = Dict{Symbol, TestSetup}()
     setup_task = @spawn begin
-        setups = Dict{Symbol, TestSetup}()
         for (name, setup) in setup_channel
-            if haskey(setups, name)
+            if haskey($setups, name)
                 @warn "Encountered duplicate @testsetup with name: `$name`. Replacing..."
             end
-            setups[name] = setup
+            $setups[name] = setup
         end
         return setups
     end
```
which would save us the `fetch` later on 

But i'm not sure i see how the `bind` idea relates

Are you thinking we'd `bind(setup_channel, setup_task)`? As i understand it, that would close the channel when the task terminates. But here, because the task is iterating the channel, we need the opposite: close the channel in order for the task to terminate 

Not super familiar with this so i could well be missing  something!

p.s. I _slightly_ preferred creating the Dict in the task and having to `fetch` it, as it seemed more explicit and one less bit of external state being mutated by a task (which the package already has a lot of and i sometimes find hard to track)

Reply:
I think Jacob specifically means something like:
````julia
julia> put_ch = Channel() do ch
           for v in ch
               println("got: ", v)
           end
       end
Channel{Any}(0) (empty)

julia> put!(put_ch, 10);
got: 10

julia> put!(put_ch, 100);
got: 100

julia> close(put_ch)

```
but i'm not sure yet how this would be used here; i haven't read anything yet